src/jit/flowgraph.cpp

```diff
@@ -20246,6 +20305,19 @@ void Compiler::fgDebugCheckBBlist(bool checkBBNum /* = false */, bool checkBBRef
             blockRefs = 1;
         }
 
+        // Check to see if this block is the beginning of a filter or a handler and adjust the ref count appropriately.
+        for (EHblkDsc *HBtab = compHndBBtab, *HBtabEnd = &compHndBBtab[compHndBBtabCount]; HBtab != HBtabEnd; HBtab++)
+        {
+            if (HBtab->ebdHndBeg == block)
+            {
+                blockRefs++;
+            }
+            if (HBtab->HasFilter() && (HBtab->ebdFilter == block))
+            {
+                blockRefs++;
+            }
+        }
+
         if (checkBBNum)
```

Comment:
This looks expensive. Can you instead change the assert below from:

```
        /* Check the bbRefs */
        noway_assert(!checkBBRefs || block->bbRefs == blockRefs);
```
to:
```
if (checkBBRefs)
{
   if (block->bbRefs != blockRefs)
   {
        // Check to see if this block is the beginning of a filter or a handler and adjust the ref count appropriately.
        for (EHblkDsc *HBtab = compHndBBtab, *HBtabEnd = &compHndBBtab[compHndBBtabCount]; HBtab != HBtabEnd; HBtab++)
        {
            if (HBtab->ebdHndBeg == block)
            {
                blockRefs++;
            }
            if (HBtab->HasFilter() && (HBtab->ebdFilter == block))
            {
                blockRefs++;
            }
        }

        assert(block->bbRefs == blockRefs);
    }
}
```

Thus, it is only computed if required, on demand.